Modules/pyexpat.c

```diff
@@ -1880,10 +1936,22 @@ MODULE_INITFUNC(void)
 #endif
 
     /* export using capsule */
-    capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
-    if (capi_object)
-        PyModule_AddObject(m, "expat_CAPI", capi_object);
-    return m;
+    PyObject *capi_object = PyCapsule_New(&capi, PyExpat_CAPSULE_NAME, NULL);
+    if (capi_object == NULL) {
+        goto error;
+    }
+
+    if (PyModule_AddObject(mod, "expat_CAPI", capi_object) < 0) {
+        Py_DECREF(capi_object);
+        goto error;
+    }
+
+    return mod;
+
+error:
+    Py_CLEAR(codes_dict);
+    Py_CLEAR(rev_codes_dict);
```

Comment:
Nitpick: there is no need to set these variables to NULL, since they are not used after this point.

```suggestion
    Py_XDECREF(codes_dict);
    Py_XDECREF(rev_codes_dict);
```